Flaps, Sprinkles, & Crumble

alanrtao opened the pull request:
<!--- PLEASE FOLLOW THE BELOW PR TEMPLATE -->

<!--- Provide a general summary of your changes in the Title above -->
flaps added

## Description
<!--- Describe your changes in detail -->
<!--- If this PR involves a major, project-wide change, please note that here -->
BREAKING CHANGE for rendering
1. Need to conform internal referencing between elements (reference everything to Scenes/Alan/Test
  - paper tiles
    - tile Visuals should contain "Square Size Manager" script with correct setting (the only one in the project at this moment)
    - tile Visuals/Top and Visuals/Bottom need to have...
      - Square Side (with filter and renderer in the same game object, material prototype Green/Blue, MeshGenerator on the same game object)
      - CrumbleMeshGenerator (with crumble shader and blur shader same as in reference scene, fracture mesh setting should be the only one in the project at the moment)
  - paper joints
    - Joint's PaperJoint should have Joint Renderer linked to Visuals/JointRenderer
    - Joint/Visuals' Joint Renderer should have filter and renderer referenced in the same gameobject, settings and square render settings set to the only settings assets in the project
3. Need to conform scene format
  - Duplicate Scenes/Alan/Test/VFX Editor Manager into your relevant scene and click "refresh"

## Changed Scipts
<!--- What classes have you added, deleted, or edited? -->
everything under scripts/VFX
new config assets under settins/vfx

## Changed Files
<!--- What other assets have you added, deleted, or edited? -->
crease shader by Carlos, mainly for integration
the green and blue material (the other ones are changed because the paper shader changed... but the only one I set the parameters in detail is green and blue)

## Related Task
<!--- What is the related trello task for this PR? -->
1. completed on the way: fold along crease, editor time sprinkles
2. need integrating:
  - cherry blossom sprinkles (and grass and rocks and stuff, currently multiple types of sprinkling not supported, need to edit SquareSide for that)
  - particles to mask folding

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
different folding scenarios under Scenes/Alan/Test
<!--- Include details of your testing environment, and the tests you ran -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/32473511/194447656-2e4e68ea-bc38-4475-ae26-4ca08233cb71.png)

alanrtao committed:
paper crumble and bump

alanrtao committed:
distance compute shader and crumbling material (demo settings on green)

alanrtao committed:
remore interval refresh

alanrtao committed:
edge tinting

alanrtao committed:
replace some of the magic numbers

alanrtao committed:
staging

alanrtao committed:
Merge branch 'main' into flaps

alanrtao committed:
Scene-wide refresh and GC friendly compute shaders

alanrtao committed:
joint vfx routing fix, progress bar, crumble material differentiation

alanrtao committed:
progress bar doesn't actually show up lol

alanrtao committed:
renaming

alanrtao committed:
stage

alanrtao committed:
Merge branch 'main' of https://github.com/ChaseOB/Barkane into flaps

# Conflicts:
#	.gitignore
#	Barkane/ProjectSettings/Packages/com.unity.probuilder/Settings.json

alanrtao committed:
Refactoring and changes for serialization

alanrtao committed:
Merge branch 'main' of https://github.com/ChaseOB/Barkane into flaps

alanrtao committed:
make overlap case look better

alanrtao committed:
staging

alanrtao committed:
mesh regen support

alanrtao committed:
avoid null mesh bugs

alanrtao committed:
replace joint line renderer

ChaseOB commented on Barkane/.vsconfig:
```diff
@@ -0,0 +1,6 @@
+﻿{
```
Delete changes since this is in the git ignore now

ChaseOB commented on Barkane/Assets/Scripts/VFX/JointRenderer.cs:
```diff
@@ -0,0 +1,517 @@
+        colorB2 = b2.EdgeTintedColor(settings.tintCorrection);
+    }
+
+    private static readonly float squareSize = 2;
```
should be at top of class

alanrtao replied:
refactored this to SquarerRenderSettings

alanrtao committed:
requested changes

alanrtao committed:
Incorporate particles in editor refresh

Merge branch 'main' of https://github.com/ChaseOB/Barkane into flaps

# Conflicts:
#	Barkane/Assets/Scripts/Foldable Paper/PaperJoint.cs
